hircluster.c

```diff
@@ -4252,3 +4447,38 @@ void redisClusterAsyncFree(redisClusterAsyncContext *acc) {
 
     hi_free(acc);
 }
+
+void initNodeIterator(nodeIterator *iter, redisClusterContext *cc) {
+    iter->cc = cc;
+    iter->route_version = cc->route_version;
+    dictInitIterator(&iter->di, cc->nodes);
+    iter->retries_left = 1;
+}
+
+cluster_node *nodeNext(nodeIterator *iter) {
```

Comment:
Would be nice, but since there is no api that takes a const it needs to be casted whatever its used for.
Many api's might reconnect or save a command which changes cluster_node data

Reply:
Oh. To introduce const, we could have to start bottom-up, i.e. start with the most internal stuff in hiredis. Different story.